src/runtime/wasm.ts

```diff
@@ -526,6 +527,18 @@ export class WasmParticle extends Particle {
     }
     this.innerParticle = this.exports[fn]();
     this.container.register(this, this.innerParticle);
+    // TODO(sjmiles): probably too soon: we need to render at least once, but we may have handle
+    // work pending. @shans says: if the particle has readable handles, onHandleUpdate is guaranteed
+    // to be called, otherwise we need `renderOutput` manually. Need to optimize this across all
+    // particle bases.
+    setTimeout(() => this.renderOutput(), 100);
```

Comment:
We definitely don't want this long term, but if we need it to move forward that's fine. Ultimately I think there should be a lifecycle method called when all readable handles are ready, or immediately if there are none.

Reply:
Yes, this was just a hack to work around the absence of that method. In the meantime, I got the clarification from Shane about `onHandleUpdate` guarantee. This needs repairs in various Particle impls (assuming we aren't getting the new lifecycle method right away), so I didn't want to block this PR on it.

Reply:
Ok, sounds good.